src/profile-logic/profile-data.js

```diff
@@ -169,297 +168,404 @@ type CallNodeTableAndStackMap = {
+    while (nextOldIndex === -1 && currentOldPrefix !== -1) {
+      subtreeRangeEndSorted[currentNewPrefix] = nextNewIndex;
+      const oldPrefixNextSibling = nextSibling[currentOldPrefix];
+      nextSiblingSorted[currentNewPrefix] =
+        oldPrefixNextSibling === -1 ? -1 : nextNewIndex;
+      nextOldIndex = oldPrefixNextSibling;
+      currentOldPrefix = prefix[currentOldPrefix];
+      currentNewPrefix = prefixSorted[currentNewPrefix];
+      currentDepth--;
+    }
+  }
+
+  for (let i = 0; i < stackIndexToCallNodeIndex.length; i++) {
+    stackIndexToCallNodeIndex[i] =
+      oldIndexToNewIndex[stackIndexToCallNodeIndex[i]];
+  }
+
+  return {
+    prefixSorted,
+    subtreeRangeEndSorted,
+    nextSiblingSorted,
+    depthSorted,
+    maxDepth,
+    length,
+    stackIndexToCallNodeIndex,
+  };
+}
+
+/**
+ * Used as part of creating the call node table.
+ *
+ * This function computes the remaining columns that haven't been computed by
+ * any other parts of call node table creation.
+ *
+ * We only compute these columns once we know the final size and order of the
+ * call node table, so that we can immediately put values in the right spot in
+ * the fixed-size typed array columns.
+ */
+function _computeCallNodeTableExtraColumns(
+  stackTable: StackTable,
+  frameTable: FrameTable,
+  stackIndexToCallNodeIndex: Int32Array,
+  frameTableInlinedIntoCol: Int32Array,
+  callNodeCount: number,
+  defaultCategory: IndexIntoCategoryList
+): CallNodeTableExtraColumns {
+  const stackCount = stackTable.length;
+  const stackTableCategoryCol = stackTable.category;
+  const stackTableFrameCol = stackTable.frame;
+  const stackTableSubcategoryCol = stackTable.subcategory;
+  const frameTableInnerWindowIDCol = frameTable.innerWindowID;
+  const frameTableFuncCol = frameTable.func;
+
+  const funcCol = new Int32Array(callNodeCount);
+  const categoryCol = new Int32Array(callNodeCount);
+  const subcategoryCol = new Int32Array(callNodeCount);
+  const innerWindowIDCol = new Float64Array(callNodeCount);
+  const inlinedIntoCol = new Int32Array(callNodeCount);
+
+  const haveFilled = new Uint8Array(callNodeCount);
+
+  for (let stackIndex = 0; stackIndex < stackCount; stackIndex++) {
+    const category = stackTableCategoryCol[stackIndex];
+    const subcategory = stackTableSubcategoryCol[stackIndex];
+    const frameIndex = stackTableFrameCol[stackIndex];
+    const inlinedIntoSymbol = frameTableInlinedIntoCol[frameIndex];
+
+    const callNodeIndex = stackIndexToCallNodeIndex[stackIndex];
+
+    if (haveFilled[callNodeIndex] === 0) {
+      const funcIndex = frameTableFuncCol[frameIndex];
+      funcCol[callNodeIndex] = funcIndex;
+
+      categoryCol[callNodeIndex] = category;
+      subcategoryCol[callNodeIndex] = subcategory;
+      inlinedIntoCol[callNodeIndex] = inlinedIntoSymbol;
+
+      const innerWindowID = frameTableInnerWindowIDCol[frameIndex];
+      if (innerWindowID !== null) {
+        innerWindowIDCol[callNodeIndex] = innerWindowID;
```

Comment:
nit: what do you think about adding a comment reminding that the TypedArray is initialized with "0"?